docker/linux/scripts/configure-tentacle.sh

```diff
@@ -27,203 +27,215 @@ if [ -f "$alreadyConfiguredSemaphore" ]; then
-		if [[ -z "$TargetEnvironment" ]]; then
-			echo "Please specify an environment name with the 'TargetEnvironment' environment variable" >&2
-			exit 1
-		fi
-
-		if [[ -z "$TargetRole" ]]; then
-			echo "Please specify a role name with the 'TargetRole' environment variable" >&2
-			exit 1
-		fi
+    if [[ -z "$ServerApiKey" && -z "$BearerToken" ]]; then
+        if [[ -z "$ServerPassword" || -z "$ServerUsername" ]]; then
+            echo "Please specify either an API key, a Bearer Token or a username/password with the 'ServerApiKey' or 'ServerUsername'/'ServerPassword' environment variables" >&2
+            exit 1
+        fi
+    fi
+
+    if [[ -z "$ServerUrl" ]]; then
+        echo "Please specify an Octopus Server with the 'ServerUrl' environment variable" >&2
+        exit 1
+    fi
+
+    if [[ ! -z "$TargetWorkerPool" ]]; then
+        if [[ ! -z "$TargetEnvironment" ]]; then
+            echo "The 'TargetEnvironment' environment variable is not valid in combination with the 'TargetWorkerPool' variable" >&2
+            exit 1
+        fi
+
+        if [[ ! -z "$TargetRole" ]]; then
+            echo "The 'TargetRole' environment variable is not valid in combination with the 'TargetWorkerPool' variable" >&2
+            exit 1
+        fi
+    else
+        if [[ -z "$TargetEnvironment" ]]; then
+            echo "Please specify an environment name with the 'TargetEnvironment' environment variable" >&2
+            exit 1
+        fi
+
+        if [[ -z "$TargetRole" ]]; then
+            echo "Please specify a role name with the 'TargetRole' environment variable" >&2
+            exit 1
+        fi
     fi
 
     echo " - server endpoint '$ServerUrl'"
     echo " - api key '##########'"
-  if [[ ! -z "$ServerCommsAddress" || ! -z "$ServerPort" ]]; then
-    echo " - communication mode 'Polling' (Active)"
-    if [[ ! -z "$ServerCommsAddress" ]]; then
-      echo " - server comms address $ServerCommsAddress"
-    fi
-    if [[ ! -z "$ServerPort" ]]; then
-      echo " - server port $ServerPort"
-    fi
-  else
-    echo " - communication mode 'Listening' (Passive)"
-    echo " - registered port $ListeningPort"
-  fi
-  if [[ ! -z "$TargetWorkerPool" ]]; then
-    echo " - worker pool '$TargetWorkerPool'"
-  else
-    echo " - environment '$TargetEnvironment'"
-    echo " - role '$TargetRole'"
-  fi
-  echo " - host '$PublicHostNameConfiguration'"
-  if [[ ! -z "$TargetName" ]]; then
-    echo " - name '$TargetName'"
-  fi
-  if [[ ! -z "$TargetTenant" ]]; then
-    echo " - tenant '$TargetTenant'"
-  fi
-  if [[ ! -z "$TargetTenantTag" ]]; then
-    echo " - tenant tag '$TargetTenantTag'"
-  fi
-  if [[ ! -z "$TargetTenantedDeploymentParticipation" ]]; then
-    echo " - tenanted deployment participation '$TargetTenantedDeploymentParticipation'"
-  fi
-  if [[ ! -z "$Space" ]]; then
-    echo " - space '$Space'"
-  fi
+    if [[ ! -z "$ServerCommsAddress" || ! -z "$ServerPort" ]]; then
+        if [[ ! -z $AsKubernetesTentacle ]]; then
```

Comment:
Currently all you need to do is set the `AsKubernetesTentacle` environment variable to something for  Tentacle to register itself as a tentacle. This script still sets `TentacleActive` or `TentaclePassive` communication styles for the type of connection.